aiohttp/web_app.py

```diff
@@ -380,15 +384,25 @@ async def _handle(self, request: Request) -> StreamResponse:
             handler = match_info.handler
 
             if self._run_middlewares:
-                for app in match_info.apps[::-1]:
-                    assert app.pre_frozen, "middleware handlers are not ready"
-                    for m in app._middlewares_handlers:
-                        handler = update_wrapper(partial(m, handler=handler), handler)
+                handler = Application._build_middlewares(handler, match_info.apps[::-1])
 
             resp = await handler(request)
 
         return resp
 
+    @staticmethod
```

Comment:
Personally, staticmethod feels pointless to me and I'd probably just make a separate function.